connect-examples/v2/python_payment/sqpaymentform.js

```diff
@@ -25,174 +20,181 @@ function requestCardNonce(event) {
+         */
+        validateShippingContact: function (contact) {
 
-      // Assign the nonce value to the hidden form field
-      document.getElementById('card-nonce').value = nonce;
+            var validationErrorObj;
+            /* ADD CODE TO SET validationErrorObj IF ERRORS ARE FOUND */
+            return validationErrorObj;
+        },
 
-      // POST the nonce form to the payment processing page
-      document.getElementById('nonce-form').submit();
+        /*
+         * callback function: cardNonceResponseReceived
+         * Triggered when: SqPaymentForm completes a card nonce request
+         */
+        cardNonceResponseReceived: function (errors, nonce, cardData) {
+            if (errors) {
+                // Log errors from nonce generation to the Javascript console
+                console.log("Encountered errors:");
+                errors.forEach(function (error) {
+                    console.log('  ' + error.message);
+                    alert(error.message);
+                });
+
+                return;
+            }
+            // Assign the nonce value to the hidden form field
+            document.getElementById('card-nonce').value = nonce;
+
+            // POST the nonce form to the payment processing page
+            document.getElementById('nonce-form').submit();
 
-    },
+        },
 
-    /*
-     * callback function: unsupportedBrowserDetected
-     * Triggered when: the page loads and an unsupported browser is detected
-     */
-    unsupportedBrowserDetected: function() {
-      /* PROVIDE FEEDBACK TO SITE VISITORS */
-    },
+        /*
+         * callback function: unsupportedBrowserDetected
+         * Triggered when: the page loads and an unsupported browser is detected
+         */
+        unsupportedBrowserDetected: function () {
+            /* PROVIDE FEEDBACK TO SITE VISITORS */
+        },
 
-    /*
-     * callback function: inputEventReceived
-     * Triggered when: visitors interact with SqPaymentForm iframe elements.
-     */
-    inputEventReceived: function(inputEvent) {
-      switch (inputEvent.eventType) {
-        case 'focusClassAdded':
-          /* HANDLE AS DESIRED */
-          break;
-        case 'focusClassRemoved':
-          /* HANDLE AS DESIRED */
-          break;
-        case 'errorClassAdded':
-          /* HANDLE AS DESIRED */
-          break;
-        case 'errorClassRemoved':
-          /* HANDLE AS DESIRED */
-          break;
-        case 'cardBrandChanged':
-          /* HANDLE AS DESIRED */
-          break;
-        case 'postalCodeChanged':
-          /* HANDLE AS DESIRED */
-          break;
-      }
-    },
+        /*
+         * callback function: inputEventReceived
+         * Triggered when: visitors interact with SqPaymentForm iframe elements.
+         */
+        inputEventReceived: function (inputEvent) {
+            switch (inputEvent.eventType) {
+                case 'focusClassAdded':
+                    /* HANDLE AS DESIRED */
```

Comment:
The comment style seems not very consistent. You use /* */ here and '//' other places. Is this something by design?

Reply:
This is pulled from our SqPaymentForm.js on our Doc site. If it needs to be changed, it should be changed wherever that comes from. I can change on mine, though (I would need to change the rest of the examples as well).

Reply:
Oh wait. /* is used for multiline comments, // is used for single line comments. I didn't completely understand, but I see what you mean now.